device/mellanox/x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/buffers_defaults_objects.j2

```diff
@@ -124,24 +124,55 @@
     }
 {%- endmacro %}
 
-{%- macro generate_queue_buffers(port_names_active, port_names_inactive) %}
+{%- macro generate_queue_buffers_with_extra_lossless_queues(port_names_active, port_names_extra_queues, port_names_inactive) %}
     "BUFFER_QUEUE": {
+{% set q_loop = namespace(last_valid=false) %}
 {% for port in port_names_active.split(',') %}
+{% if port not in port_names_extra_queues.split(',') %}
         "{{ port }}|3-4": {
             "profile" : "[BUFFER_PROFILE|egress_lossless_profile]"
         },
+{% endif %}
 {% endfor %}
 {% for port in port_names_active.split(',') %}
+{% if port not in port_names_extra_queues.split(',') %}
         "{{ port }}|0-2": {
             "profile" : "[BUFFER_PROFILE|q_lossy_profile]"
         },
+{% endif %}
 {% endfor %}
 {% for port in port_names_active.split(',') %}
+{% if port not in port_names_extra_queues.split(',') %}
         "{{ port }}|5-6": {
             "profile" : "[BUFFER_PROFILE|q_lossy_profile]"
         }{% if not loop.last %},{% endif %}
 
+{% set q_loop.last_valid = true %}
+{% else %}
+{% set q_loop.last_valid = false %}
+{% endif %}
+{% endfor %}
+{% if port_names_extra_queues|length > 0 %}
+{% if q_loop.last_valid %},{% endif %}
+{% for port in port_names_extra_queues.split(',') %}
+        "{{ port }}|0-1": {
+            "profile" : "[BUFFER_PROFILE|q_lossy_profile]"
+        },
+        "{{ port }}|2-4": {
+            "profile" : "[BUFFER_PROFILE|egress_lossless_profile]"
+        },
+        "{{ port }}|5": {
+            "profile" : "[BUFFER_PROFILE|q_lossy_profile]"
+        },
+        "{{ port }}|6": {
+            "profile" : "[BUFFER_PROFILE|egress_lossless_profile]"
+        },
+        "{{ port }}|7": {
+            "profile" : "[BUFFER_PROFILE|q_lossy_profile]"
+        }{% if not loop.last %},{% endif %}
+
 {% endfor %}
+{% endif %}
```

Comment:
For L196,209 don't we need to add the correct zero buffer profile when there are extra lossless queues?

Reply:
So as long as any zero profile (lossless or lossy) is attached to that queue, we should be able to reclaim?

Reply:
Yes